azurelinuxagent/common/protocol/wire.py

```diff
@@ -876,6 +880,25 @@ def get_gafamily_manifest(self, vmagent_manifest, goal_state):
 
         raise ProtocolError("Failed to retrieve GAFamily manifest")
 
+    def _remove_stale_agent_manifest(self, family, incarnation):
+        """
+        The incarnation number can reset at any time, which means there
+        could be a stale agentsManifest on disk.  Stale files are cleaned
+        on demand as new goal states arrive from WireServer. If the stale
+        file is not removed agent upgrade may be delayed.
+
+        :param family: GA family, e.g. Prod or Test
+        :param incarnation: incarnation of the current goal state
+        """
+        fn = AGENTS_MANIFEST_FILE_NAME.format(
+            family,
+            incarnation)
+
+        agent_manifest = os.path.join(conf.get_lib_dir(), fn)
```

Comment:
Is there a reason to duplicate the filename generation logic rather than pass in the final `local_file` name?

Reply:
I don't see the duplication.  There are two file names: MANIFEST_FILE_NAME and AGENTS_MANIFEST_FILE_NAME.

I am probably missing what you're seeing.

Reply:
Nope, you're right. I misread the constants used.